geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZIncrByIntegrationTest.java

```diff
@@ -0,0 +1,290 @@
+            .hasMessageContaining(RedisConstants.ERROR_NOT_A_VALID_FLOAT);
+  }
+
+  @Test
+  public void shouldError_givenNaNIncrement() {
+    String nanIncrement = "NaN";
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+
+    assertThatThrownBy(() -> jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY,
+        nanIncrement, STRING_MEMBER))
+            .hasMessageContaining(RedisConstants.ERROR_NOT_A_VALID_FLOAT);
+  }
+
+  /************* infinity *************/
+  @Test
+  public void shouldSetScoreToInfinity_ifIncrementWouldExceedMaxValue() {
+    double increment = Double.MAX_VALUE / 2;
+
+    jedis.zadd(STRING_KEY, Double.MAX_VALUE, STRING_MEMBER);
+    jedis.zincrby(STRING_KEY, increment, STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, -Double.MAX_VALUE, STRING_MEMBER);
+    jedis.zincrby(STRING_KEY, -increment, STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(NEGATIVE_INFINITY);
+  }
+
+  @Test
+  public void shouldSetScoreToInfinity_givenInfiniteIncrement() {
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    assertThat(jedis.zincrby(STRING_KEY, POSITIVE_INFINITY, STRING_MEMBER))
+        .isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    assertThat(jedis.zincrby(STRING_KEY, NEGATIVE_INFINITY, STRING_MEMBER))
+        .isEqualTo(NEGATIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "inf", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "+inf", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "-inf", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(NEGATIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "Infinity", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "+Infinity", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "-Infinity", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(NEGATIVE_INFINITY);
+  }
+
+  @Test
+  public void nullRedisSetShouldCreateNonExistentMember_withIncrementOfPositiveInfinity() {
```

Comment:
The name of this test is a little confusing to me. Is it intended to test that `incrby` called on a null set should behave as expected? Because if so, the call to `zadd` should be omitted, since that creates the set, meaning that the code in `NullRedisSortedSet.zincrby()` is never executed.

Reply:
Line 145 uses "something" as the member, then 147 uses the `MEMBER` constant. I have to do that first zadd to initialize the key and sorted set, then the second one creates the member

Reply:
Thanks for the clarification